### ISSUE TITLE
Sqlalchemy error when updating csw record with a repo filter

### ISSUE THREAD

**Issue by sarametz** [geopython/pycsw#298]
I'm using pycsw 1.10 in django with some complex filtering and have come across a sqlalchemy error to do with the update() method and synchronize_session parameter. I've managed to reproduce the error in a simplified setup and it seems to be when updating a csw record with repository.filter set in the pycsw config.  
The error occurs when running the wsgi server with `repository.filter=type = 'http://purl.org/dc/dcmitype/Dataset'`, `manager.transactions=true`,`manager.allowed_ips=*.*.*.*` and database `pycsw/tests/suites/cite/data/records.db` if I attempt to update one of the preexisting records I get the response:
```xml
<ows:ExceptionReport xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:inspire_common="http://inspire.ec.europa.eu/schemas/common/1.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:dct="http://purl.org/dc/terms/" xmlns:ows="http://www.opengis.net/ows" xmlns:apiso="http://www.opengis.net/cat/csw/apiso/1.0" xmlns:gml="http://www.opengis.net/gml" xmlns:dif="http://gcmd.gsfc.nasa.gov/Aboutus/xml/dif/" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:gco="http://www.isotc211.org/2005/gco" xmlns:gmd="http://www.isotc211.org/2005/gmd" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:srv="http://www.isotc211.org/2005/srv" xmlns:ogc="http://www.opengis.net/ogc" xmlns:fgdc="http://www.opengis.net/cat/csw/csdgm" xmlns:inspire_ds="http://inspire.ec.europa.eu/schemas/inspire_ds/1.0" xmlns:csw="http://www.opengis.net/cat/csw/2.0.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:os="http://a9.com/-/spec/opensearch/1.1/" xmlns:soapenv="http://www.w3.org/2003/05/soap-envelope" xmlns:sitemap="http://www.sitemaps.org/schemas/sitemap/0.9" language="en-US" version="1.2.0" xsi:schemaLocation="http://www.opengis.net/ows http://schemas.opengis.net/ows/1.0.0/owsExceptionReport.xsd">
<ows:Exception exceptionCode="NoApplicableCode" locator="update">
<ows:ExceptionText>Transaction (update) failed: 'InvalidRequestError' object has no attribute 'orig'.</ows:ExceptionText>
</ows:Exception>
</ows:ExceptionReport>
``` 
If I print the error at line 318 of `pycsw/repository.py` I get   
`Could not evaluate current criteria in Python. Specify 'fetch' or False for the synchronize_session parameter`

Altering line 315/316 in `pycsw/repository.py` from `self._get_repo_filter(self.session.query(self.dataset)).filter_by( identifier=identifier).update(update_dict)`
to
`self._get_repo_filter(self.session.query(self.dataset)).filter_by( identifier=identifier).update(update_dict, synchronize_session=False)` seems to fix the problem

pycsw system profile
    --------------------
    Python version: sys.version_info(major=2, minor=7, micro=5, releaselevel='final', serial=0)
    os: darwin
    SQLAlchemy: 0.9.8
    Shapely: 1.3.1
    lxml: 3.3.5
    libxml2: (2, 9, 0)
    pyproj: 1.9.3
    OWSLib: 0.8.8

**Comment by tomkralidis** [comment]
@sarametz thanks for the info and awesome report!  If possible, a pull request with the change will be helpful to trigger our travis-ci setup/tests.

**Comment by tomkralidis** [comment]
looking deeper, we use `synchronize_session='fetch'` in https://github.com/geopython/pycsw/blob/master/pycsw/repository.py#L348 and https://github.com/geopython/pycsw/blob/master/pycsw/repository.py#L348.  Does `synchronize_session='fetch'` work for you too? Might be good (either `fetch` or `False`) to be consistent in the update() function.

**Comment by sarametz** [comment]
Yeah, fetch also works for me

**Comment by tomkralidis** [comment]
@sarametz merged PR #298 and applied to 1.10 branch as well.  Thanks much for this valuable fix :+1:
